Qitom/python/pythonItom.cpp

```diff
@@ -161,6 +161,66 @@ PyObject* PythonItom::PyNewScript(PyObject* /*pSelf*/, PyObject* /*pArgs*/)
     }
 }
 
+//-------------------------------------------------------------------------------------
+PyDoc_STRVAR(pyLog_doc, "log(msg) \n\
+\n\
+Writes a message to the log file if itom was started with the \"log\" option.\n\
+\n\
+The message will be prepended with [Python  <date>].\n\
+\n\
+Parameters \n\
+---------- \n\
+msg : str \n\
+    The message to be written to the log file. \n\
+\n\
+Raises \n\
+------ \n\
+RuntimeWarning \n\
+    if itom was not started with the \"log\" option.");
+PyObject* PythonItom::PyLog(PyObject* /*pSelf*/, PyObject* pArgs)
+{
+    ito::RetVal retVal(ito::retOk);
+
+    if (PyTuple_Size(pArgs) != 1)
+    {
+        retVal += ito::RetVal(ito::retError, 0, "Wrong number of arguments");
+    }
+    QString text;
+    if (!retVal.containsWarningOrError())
+    {
+        PyObject* tempObj = PyTuple_GetItem(pArgs, 0);
+        if (PyUnicode_Check(tempObj))
+        {
+            bool ok = false;
+            text = PythonQtConversion::PyObjGetString(tempObj, false, ok);
+        }
+        else
+        {
+            retVal +=
+                ito::RetVal(ito::retError, 0, "Argument has to be a string! Wrong argument type!");
+        }
+    }
+    QObject* logger;
+    if (!retVal.containsWarningOrError())
+    {
+        logger = AppManagement::getLogger();
+        if (logger == nullptr)
+        {
+            retVal += ito::RetVal(ito::retWarning, 0, "Logger not available");
```

Comment:
If no logger has been created by purpose, I would create a noop method out of this (no warning if logger not available).